File: src/main/java/com/github/kory33/signvote/exception/VotePointNotVotedException.java
package com.github.kory33.signvote.exception;

import com.github.kory33.signvote.vote.VotePoint;
import com.github.kory33.signvote.session.VoteSession;
import lombok.EqualsAndHashCode;
import lombok.Value;

import java.util.UUID;

/**
 * Represents an exception thrown when a cancel of non-existent vote has been requested.
 */
@EqualsAndHashCode(callSuper = false)
@Value
public class VotePointNotVotedException extends Exception {
    UUID voterUUID;
    VotePoint votePoint;
    VoteSession session;
}

File: src/main/java/com/github/kory33/signvote/exception/VotePointNotVotedException.kt
package com.github.kory33.signvote.exception

import com.github.kory33.signvote.session.VoteSession
import com.github.kory33.signvote.vote.VotePoint

import java.util.UUID

/**
 * Represents an exception thrown when a cancel of non-existent vote has been requested.
 */
data class VotePointNotVotedException
constructor(val voterUUID: UUID, val votePoint: VotePoint, val session: VoteSession) : Exception()
